File: app/src/main/java/com/example/dieunnph15766_ass/fragment/IncomeTypeFragment.java
package com.example.dieunnph15766_ass.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.dieunnph15766_ass.R;
import com.example.dieunnph15766_ass.dialog.CustomDialog;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class IncomeTypeFragment extends Fragment implements CustomDialog.CustomDialogListener {
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = LayoutInflater.from(container.getContext()).inflate(R.layout.fragment_income_type, null);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        FloatingActionButton fab = (FloatingActionButton) view.findViewById(R.id.floating_action_button_add_new_income_type);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                CustomDialog dialog = new CustomDialog();
                dialog.show(getParentFragmentManager(), "TAG");

            }
        });
    }

    @Override
    public void applyText(String name) {
        Toast.makeText(getActivity().getBaseContext(), name, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/dieunnph15766_ass/fragment/IncomeTypeFragment.kt
package com.example.dieunnph15766_ass.fragment

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.preference.PreferenceManager
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentResultListener
import com.example.dieunnph15766_ass.R
import com.example.dieunnph15766_ass.activity.MainActivity
import com.example.dieunnph15766_ass.database.Database
import com.example.dieunnph15766_ass.database.IncomeTypeDB
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.example.dieunnph15766_ass.dialog.CustomDialog
import com.example.dieunnph15766_ass.model.IncomeType

class IncomeTypeFragment : Fragment(), CustomDialog.OnInputSelected {
    lateinit var textResult: TextView

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return LayoutInflater.from(container!!.context)
            .inflate(R.layout.fragment_income_type, null)

    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        textResult = view.findViewById<TextView>(R.id.textview_recieve_from_custom_dialog)
        textResult.text = "Text example"
        val fab =
            view.findViewById<View>(R.id.floating_action_button_add_new_income_type) as FloatingActionButton
        fab.setOnClickListener {
            var dialog = CustomDialog()
            dialog.setTargetFragment(this, 0)
            dialog.show(parentFragmentManager, "Tag")
        }

    }

    override fun setText(text: String) {
        // Insert into database
        val database = Database(parentFragment?.activity)
        val incomeTypeDB = IncomeTypeDB(database)
        textResult.text = text
        val username = PreferenceManager.getDefaultSharedPreferences(context).getString("USERNAME", "")
        if(incomeTypeDB.newIncomeType(IncomeType(null, text, username))) {
            Toast.makeText(context, "Add successfully", Toast.LENGTH_SHORT).show()
        } else  {
            Toast.makeText(context, "Add Failed", Toast.LENGTH_SHORT).show()
        }
    }
}